packages/aws-cdk/lib/api/hotswap/monitor/logs-monitor.ts

```diff
@@ -0,0 +1,226 @@
+   * in parallel
+   */
+  private async readNewEvents(): Promise<void[]> {
+    const promises: Array<Promise<void>> = [];
+    for (let group of this.logGroups) {
+      promises.push(this.readEventsFromLogGroup(group));
+    }
+    return Promise.all(promises);
+  }
+
+  /**
+   * Reads all new log events from a CloudWatch Log Group
+   * starting when the hotswap was triggered
+   *
+   * Only prints out events that have not been printed already
+   */
+  private async readEventsFromLogGroup(logGroupName: string): Promise<void> {
+    // this should not be possible
+    if (!this.sdk) {
+      return;
+    }
+    try {
+      let finished = false;
+      let nextToken: string | undefined;
+      while (!finished) {
+        const response = await this.sdk!.cloudwatchLogs().filterLogEvents({
+          logGroupName: logGroupName,
+          nextToken,
+          startTime: this.startTime,
+        }).promise();
+        const eventPage = response.events ?? [];
+
+
+        for (const event of eventPage) {
+          if (event.eventId && this.activity.has(event.eventId)) {
+          } else {
+            if (event.eventId) this.activity.add(event.eventId);
+            if (event.message) {
+              const ingestionTime: Date = event.ingestionTime ?
+                new Date(event.ingestionTime) : new Date();
+              this.printer.print({
+                message: event.message,
+                logGroup: logGroupName,
+                logStream: event.logStreamName,
+                ingestionTime,
+              });
+            }
+
+          }
+        }
+
+        nextToken = response.nextToken;
+        if (nextToken === undefined) {
+          finished = true;
+        }
+      }
+    } catch (e) {
+      // with Lambda functions the CloudWatch is not created
+      // until something is logged, so just keep polling until
+      // there is somthing to find
+      if (e.code === 'ResourceNotFoundException') {
+        return;
+      }
+      throw e;
+    }
+  }
+
+}
+
+interface PrinterProps {
+  /**
+   * Stream to write to
+   */
+  readonly stream: NodeJS.WriteStream;
+}
+
+export interface IEventPrinter {
+  readonly updateSleep: number;
+
+  print(event: CloudWatchLogEvent): void;
+}
+
+abstract class EventPrinterBase {
+  public readonly updateSleep: number = 5_000;
+  protected readonly stream: NodeJS.WriteStream;
+
+  constructor(protected readonly props: PrinterProps) {
+    this.stream = props.stream;
+  }
+
+  public abstract print(event: CloudWatchLogEvent): void;
+}
+
+export class EventPrinter extends EventPrinterBase {
+  constructor(props: PrinterProps) {
+    super(props);
+  }
+
+  public print(event: CloudWatchLogEvent): void {
+    this.stream.write(
+      util.format(
+        '%s %s %s %s',
+        colors.blue(event.logGroup),
+        (event.logStream ? colors.blue(event.logStream) : ''),
+        colors.yellow(event.ingestionTime.toLocaleTimeString()),
+        event.message,
+      ),
+    );
```

Comment:
Formatting reads a little strange here:

```suggestion
    this.stream.write(util.format('%s %s %s %s',
      colors.blue(event.logGroup),
      event.logStream ? colors.blue(event.logStream) : '',
      colors.yellow(event.ingestionTime.toLocaleTimeString()),
      event.message));
```